src/containers/pages/FocusInvestigation/map/active/index.tsx

```diff
@@ -161,52 +163,28 @@ class SingleActiveFIMap extends React.Component<
     super(props);
   }
 
-  public async componentDidMount() {
-    const {
-      fetchGoalsActionCreator,
-      fetchJurisdictionsActionCreator,
-      fetchPlansActionCreator,
-      fetchStructuresActionCreator,
-      fetchTasksActionCreator,
-      plan,
-      supersetService,
-    } = this.props;
-
-    if (plan && plan.plan_id) {
-      /** define superset filter params for jurisdictions */
-      const jurisdictionsParams = superset.getFormData(SUPERSET_MAX_RECORDS, [
-        { comparator: plan.jurisdiction_id, operator: '==', subject: 'jurisdiction_id' },
-      ]);
-      await supersetService(SUPERSET_JURISDICTIONS_SLICE, jurisdictionsParams).then(
-        (result: Jurisdiction[]) => fetchJurisdictionsActionCreator(result)
-      );
-      /** define superset params for filtering by plan_id */
-      const supersetParams = superset.getFormData(SUPERSET_MAX_RECORDS, [
-        { comparator: plan.plan_id, operator: '==', subject: 'plan_id' },
+  public componentDidMount() {
+    if (!this.props.plan) {
+      /**
+       * Fetch plans again because on reloading the page, the state does not persist
+       * the plans
+       */
+      const { supersetService, fetchPlansActionCreator } = this.props;
+      const supersetParams = superset.getFormData(2000, [
+        { comparator: InterventionType.FI, operator: '==', subject: PLAN_INTERVENTION_TYPE },
       ]);
-      /** define superset params for goals */
-      const goalsParams = superset.getFormData(
-        SUPERSET_MAX_RECORDS,
-        [{ comparator: plan.plan_id, operator: '==', subject: 'plan_id' }],
-        { action_prefix: true }
-      );
-      /** Implement Ad hoc Queries since jurisdictions have no plan_id */
-      await supersetService(SUPERSET_STRUCTURES_SLICE, jurisdictionsParams).then(
-        (structuresResults: Structure[]) => {
-          fetchStructuresActionCreator(structuresResults);
-        }
-      );
-      await supersetService(SUPERSET_PLANS_SLICE, jurisdictionsParams).then((result2: Plan[]) => {
-        fetchPlansActionCreator(result2);
-      });
-      await supersetService(SUPERSET_GOALS_SLICE, goalsParams).then((result3: Goal[]) => {
-        fetchGoalsActionCreator(result3);
-      });
-      await supersetService(SUPERSET_TASKS_SLICE, supersetParams).then((result4: Task[]) => {
-        fetchTasksActionCreator(result4);
-      });
+      supersetService(SUPERSET_PLANS_SLICE, supersetParams)
+        .then((result: Plan[]) => fetchPlansActionCreator(result))
+        .catch(error => displayError(error));
+    } else {
+      /**
+       * Plans are available in state since the plans were fetched by the
+       * the previous view
+       */
+      this.fetchData().catch(error => displayError(error));
```

Comment:
Does this actually catch the errors if no promises are being returned in `this.fetchData()`?

Reply:
@cKellyDesign Tslint raises a flag if there's no catch block
![Screenshot from 2020-02-21 18-03-48](https://user-images.githubusercontent.com/19818819/75045581-96eaed80-54d4-11ea-80c8-34b188f7f551.png)

Reply:
@kelvin-muchiri I definitely think it should have a `catch` block, I just want to make sure that if an error is encountered within `fetchData()`, it's thrown correctly and get's caught by this `catch`. Can we add a test for such a scenario?

Reply:
God bless this more aggressive `tslist` :) #amiright? :)

Reply:
> Does this actually catch the errors if no promises are being returned in `this.fetchData()`?

Apparently we have to add a `then` or `catch` block else the compiler thinks we forgot to handle our async code. TSlint was raising errors (https://stackoverflow.com/a/43981878/3247914)

Reply:
> @kelvin-muchiri I definitely think it should have a `catch` block, I just want to make sure that if an error is encountered within `fetchData()`, it's thrown correctly and get's caught by this `catch`. Can we add a test for such a scenario?

Added tests